pandas/core/groupby/generic.py

```diff
@@ -318,13 +319,18 @@ def _aggregate_multiple_funcs(self, arg):
                 obj = copy.copy(obj)
                 obj._reset_cache()
                 obj._selection = name
-            results[name] = obj.aggregate(func)
+            results.append(Result(name, obj.aggregate(func)))
 
-        if any(isinstance(x, DataFrame) for x in results.values()):
+        if any(isinstance(x.aggregation, DataFrame) for x in results):
             # let higher level handle
-            return results
+            return dict(results)
 
-        return DataFrame(results, columns=columns)
+        # If there are multiple aggregations with the same name, we need to pass
+        # them to the constructor with different keys.
+        df = DataFrame({n: x.aggregation for n, x in enumerate(results)}).rename(
```

Comment:
We have similar functionality already built in to do this. If you keep `results` as a dict and use a `base.OutputKey` with the results as a value you can ultimately just call `self._wrap_aggregated_output`

Reply:
@WillAyd nice, thanks!

I've updated the PR with that instead, is there a better way of writing this line:
```
{key.label: value for key, value in results.items()}
```
?